girder/esimmon_plugin/esimmon_plots/esimmonfile.py

```diff
@@ -0,0 +1,49 @@
+import
+from girder.api import access
+from girder.api.describe import Description, autoDescribeRoute
+from girder.api.rest import getCurrentUser
+
+from girder.utility.path import getResourcePath
+
+from girder.models.collection import Collection
+from girder.models.folder import Folder
+
+def _filter_items(folders, user, query, matches, parent):
+    for folder in folders:
+        if not
+        items = list(Folder().childItems(folder, parentType='folder', user=user))
+        for item in items:
+            full_path = getResourcePath('item', item, user)
+            trimmed_path = full_path.split(parent['name'])[1]
+            if query in trimmed_path:
+                matches.append({'text': trimmed_path, 'value': item})
+        folders = list(Folder().childFolders(folder, parentType='folder', user=user))
+        if folders:
+            _filter_items(folders, user, query, matches, parent)
+    return matches
+
+@access.public
+@autoDescribeRoute(
+  Description('Search for items that are children of the given resource.')
+  .param('id', 'The ID of the resource.', paramType='path')
+  .param('type', 'The type of the resource (must be collection or folder).')
+  .param('q', 'The search query.', dataType='string', required=False)
+)
+def find_parameters(id, type, q):
```

Comment:
If this is not specific to "parameters" we should probably rename?

Reply:
Renamed to just `find_items` since it is just finding all items in the sub-tree.